src/inspector_socket.cc

```diff
@@ -159,18 +161,16 @@ static std::vector<char> encode_frame_hybi17(const char* message,
   return frame;
 }
 
-static ws_decode_result decode_frame_hybi17(const char* buffer_begin,
-                                            size_t data_length,
+static ws_decode_result decode_frame_hybi17(const std::vector<char>& buffer,
                                             bool client_frame,
                                             int* bytes_consumed,
                                             std::vector<char>* output,
                                             bool* compressed) {
   *bytes_consumed = 0;
-  if (data_length < 2)
+  if (buffer.size() < 2)
     return FRAME_INCOMPLETE;
 
-  const char* p = buffer_begin;
-  const char* buffer_end = p + data_length;
+  std::vector<char>::const_iterator p = buffer.begin();
```

Comment:
I'd be okay with writing this as `auto p = buffer.cbegin()` (but maybe call it `it` to make it clear it's an iterator, not a pointer.)